src/core/symbology-ng/qgscategorizedsymbolrenderer.cpp

```diff
@@ -136,9 +136,20 @@ void QgsRendererCategory::toSld( QDomDocument &doc, QDomElement &element, QgsStr
   ruleElem.appendChild( descrElem );
 
   // create the ogc:Filter for the range
-  QString filterFunc = QStringLiteral( "%1 = '%2'" )
-                       .arg( attrName.replace( '\"', QLatin1String( "\"\"" ) ),
-                             mValue.toString().replace( '\'', QLatin1String( "''" ) ) );
+  QString filterFunc;
+  if ( mValue.isNull() || mValue.toString().isEmpty() )
```

Comment:
@m-kuhn what do you mean with more control on "" and NULL using rule based? with this PR they are managed in the same way. Are you thinking to split the "or" with two different rules?